### PR TITLE
Use simple custom CORS header middleware

### PR DESCRIPTION
**Pull request by jseppi**
This PR is an attempt to mitigate #1307.

I've removed the `django-cors-headers` module and replaced it with an extremely simple custom middleware called `ApiCorsMiddleware`. This new middleware adds the CORS `'Access-Control-Allow-Origin: *'` header to all responses to requests that start with the `API_HOST` prefix.

### CODE REVIEW

File: hourglass/middleware.py [hunk 1]
```diff
@@ -52,3 +52,14 @@ def __init__(self):
...
         super().__init__()
+
+
+class ApiCorsMiddleware:
+    '''
+    Middleware to add CORS headers to all responses from requests to API_HOST
+    paths.
+    '''
+    def process_response(self, request, response):
+        if request.path.startswith(settings.API_HOST):
```
Comment (toolness) [inline hourglass/middleware.py]: wait won't `API_HOST` be an absolute URL in production though? GArgh I'm confused

Comment (jseppi) [inline hourglass/middleware.py]: hrm, yeah, you are right >.<

Comment (jseppi) [inline hourglass/middleware.py]: I misunderstood what that setting was for. I remember now.

Comment (jseppi) [inline hourglass/middleware.py]: I changed it to just have `/api` hardcoded in the check

**Comment by toolness** [comment]
Er wait, so while `django-cors-headers` does seem overkill for our needs, do you think it was actively causing #1307?  

Regarding your comment in #1307:

> Hrm, new theory: CloudFront isn't stripping the `Access-Control-Allow-Origin` header, it isn't being added by the `django-cors-headers` middleware due to `HTTP_ORIGIN` not being set (see logic at https://github.com/ottoyiu/django-cors-headers/blob/862c68d1eb65455cf0051c4f124a4eb4341b68f2/corsheaders/middleware.py#L99). Maybe. I have no idea. ugh.

I believe that `HTTP_ORIGIN` (i.e. the `Origin:` header) is *always* sent by any browser that supports CORS, when it's aware that it's sending a cross-origin request.

Oh wait.  Um, I guess in this case it's really the API umbrella instance that's talking to our `/api/` endpoint, isn't it?  And *that* might not be setting an `Origin:` header... Ohhh man I am confused.

**Comment by jseppi** [comment]
I actually still don't know what was causing the problem. We can hold off on this PR until we figure it out, but I wanted to have something ready in case I guess.

**Comment by jseppi** [comment]
Newest theory: https://github.com/18F/calc/issues/1307#issuecomment-276713325

**Comment by jseppi** [comment]
Closing because I _think_ we isolated the problem down to a (suprise) caching issue (ref https://github.com/18F/calc/issues/1307#issuecomment-276713325).

**Pull request closed without merging**